modules/liquidity-pools/margin/src/lib.rs

```diff
@@ -318,20 +319,25 @@ impl<T: Trait> MarginProtocolLiquidityPools<T::AccountId> for Module<T> {
 	fn get_swap_rate(pool_id: LiquidityPoolId, pair: TradingPair, is_long: bool) -> Fixed128 {
 		let max_swap = T::MaxSwap::get();
 		let swap_rate = Self::swap_rate(pair).unwrap_or_default();
-		let adjust_rate: Fixed128;
-		if is_long {
-			adjust_rate = swap_rate
-				.long
-				.saturating_sub(Self::additional_swap_rate(pool_id).unwrap_or_default());
-		} else {
-			adjust_rate = swap_rate
-				.short
-				.saturating_sub(Self::additional_swap_rate(pool_id).unwrap_or_default());
-		}
+		let additional_swap_rate = Self::additional_swap_rate(pool_id).unwrap_or_default();
+		let one = Fixed128::from_natural(1);
+		let adjust_rate = {
+			if is_long {
+				// swap_rate = long_rate * (1 + additional_swap_rate)
```

Comment:
after bit more thinking, I don't think this is the correct formula because long can have negative swap and short can have positive swap.

let me check with Antonia